Userland/Libraries/LibWeb/UIEvents/WheelEvent.h

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2022, the SerenityOS developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/TypeCasts.h>
+#include <LibWeb/UIEvents/EventModifier.h>
+#include <LibWeb/UIEvents/MouseEvent.h>
+#include <LibWeb/UIEvents/UIEvent.h>
+
+namespace Web::UIEvents {
+
+struct WheelEventInit : public MouseEventInit {
```

Comment:
The constants from the IDL should be reproduced here as constexpr varaibles
```
    const unsigned long DOM_DELTA_PIXEL = 0x00;
    const unsigned long DOM_DELTA_LINE  = 0x01;
    const unsigned long DOM_DELTA_PAGE  = 0x02;
```

Reply:
Reading the spec, it looks like they could actually be an `enum`.

> deltaMode, of type [unsigned long](https://webidl.spec.whatwg.org/#idl-unsigned-long), defaulting to 0
    Initializes the [deltaMode](https://www.w3.org/TR/uievents/#dom-wheelevent-deltamode) attribute on the [WheelEvent](https://www.w3.org/TR/uievents/#wheelevent) object to the **enumerated values 0, 1, or 2,** which represent the amount of pixels scrolled ([DOM_DELTA_PIXEL](https://www.w3.org/TR/uievents/#dom-wheelevent-dom_delta_pixel)), lines scrolled ([DOM_DELTA_LINE](https://www.w3.org/TR/uievents/#dom-wheelevent-dom_delta_line)), or pages scrolled ([DOM_DELTA_PAGE](https://www.w3.org/TR/uievents/#dom-wheelevent-dom_delta_page)) if the [rotation](https://www.w3.org/TR/uievents/#rotation) of the wheel would have resulted in scrolling.
    
Something like 
```c++
enum WheelDeltaMode : unsigned long
{
    DOM_DELTA_PIXEL = 0,
    DOM_DELTA_LINE = 1,
    DOM_DELTA_PAGE = 2,
};
```

delta_mode() could then be `return to_underlying(m_delta_mode);`